cpp/bench/sg/fil.cu

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2019-2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuml/fil/fil.h>
+#include <cuml/tree/algo_helper.h>
+#include <treelite/c_api.h>
+#include <treelite/tree.h>
+#include <cuml/common/logger.hpp>
+#include <cuml/cuml.hpp>
+#include <utility>
+#include "benchmark.cuh"
+
+namespace ML {
+namespace Bench {
+namespace fil {
+
+struct Params {
+  DatasetParams data;
+  RegressionParams blobs;
+  ModelHandle model;
+  bool predict_proba;
+};
+
+class FIL : public RegressionFixture<float> {
+ public:
+  FIL(const std::string& name, const Params& p)
+    : RegressionFixture<float>("FIL", p.data, p.blobs),
+      model(p.model),
+      predict_proba(p.predict_proba) {
+    this->SetName(name.c_str());
+  }
+
+ protected:
+  void runBenchmark(::benchmark::State& state) override {
+    if (!this->params.rowMajor) {
+      state.SkipWithError("FIL only supports row-major inputs");
+    }
+    auto& handle = *this->handle;
+    auto stream = handle.getStream();
+    for (auto _ : state) {
+      MLCommon::Bench::CudaEventTimer timer(state, nullptr, 0, stream);
+      ML::fil::predict(handle, forest, this->data.y, this->data.X,
+                       this->params.nrows, predict_proba);
+    }
```

Comment:
Can you check the latest `cpp/bench/common/benchmark.cuh` file for a helper `loopOnState` method that can further simplify this?